auth/auth/auth.py

```diff
@@ -229,6 +296,12 @@ async def callback(request):
         if domain != ORGANIZATION_DOMAIN:
             raise web.HTTPUnauthorized()
 
+        try:
+            await check_valid_new_user(db, username, login_id, is_developer=False, is_service_account=False)
+        except AuthUserError as e:
+            set_message(session, e.message, 'error')
+            return web.HTTPFound(deploy_config.external_url('notebook', ''))
```

Comment:
I will fix the notebook redirect in a future PR. I just wanted it to be consistent for now.